Sources/KSCrashRecording/include/KSCrashReportStore.h

```diff
@@ -0,0 +1,140 @@
+    KSCDeleteNever,
+    KSCDeleteOnSucess,
+    KSCDeleteAlways
+} NS_SWIFT_NAME(DeleteBehavior);
+
+NS_SWIFT_NAME(CrashReportStore)
+@interface KSCrashReportStore : NSObject
+
+/** The default folder name inside the KSCrash install path that is used for report store.
+ */
+@property(nonatomic, class, copy, readonly) NSString *defaultInstallSubfolder;
+
+- (instancetype)init NS_UNAVAILABLE;
++ (instancetype)new NS_UNAVAILABLE;
+
+/** The report store with the default configuration.
+ *
+ * @param error If an error occurs, upon return contains an NSError object that
+ *               describes the problem.
+ *
+ * @return The default report store oe nil if an error occurred.
+ */
++ (nullable instancetype)defaultStoreWithError:(NSError **)error;
+
+/** The report store with the given configuration.
+ * If the configuration is nil, the default configuration will be used.
+ *
+ * @param configuration The configuration to use.
+ * @param error If an error occurs, upon return contains an NSError object that
+ *
+ * @return The report store or nil if an error occurred.
+ */
++ (nullable instancetype)storeWithConfiguration:(nullable KSCrashReportStoreConfiguration *)configuration
+                                          error:(NSError **)error;
+
+#pragma mark - Configuration
+
+/** The report sink where reports get sent.
+ * This MUST be set or else the reporter will not send reports (although it will
+ * still record them).
+ *
+ * Note: If you use an installation, it will automatically set this property.
+ *       Do not modify it in such a case.
+ */
+@property(nonatomic, readwrite, strong, nullable) id<KSCrashReportFilter> sink;
+
+/** What to do after sending reports via sendAllReportsWithCompletion:
+ *
+ * - Use KSCDeleteNever if you will manually manage the reports.
+ * - Use KSCDeleteAlways if you will be using an alert confirmation (otherwise it
+ *   will nag the user incessantly until he selects "yes").
+ * - Use KSCDeleteOnSuccess for all other situations.
+ *
+ * Default: KSCDeleteAlways
+ */
+@property(nonatomic, assign) KSCDeleteBehavior deleteBehaviorAfterSendAll;
+
+#pragma mark - Reports API
+
+/** Get all unsent report IDs. */
+@property(nonatomic, readonly, strong) NSArray<NSNumber *> *reportIDs;
+
+/** Send all outstanding crash reports to the current sink.
+ * It will only attempt to send the most recent 5 reports. All others will be
+ * deleted. Once the reports are successfully sent to the server, they may be
+ * deleted locally, depending on the property "deleteAfterSendAll".
+ *
+ * @note A call of `setupReportStoreWithPath:error:` or `installWithConfiguration:error:` is required
+ *       before working with crash reports.
+ * @note Property "sink" MUST be set or else this method will call `onCompletion` with an error.
+ *
+ * @param onCompletion Called when sending is complete (nil = ignore).
+ */
+- (void)sendAllReportsWithCompletion:(nullable KSCrashReportFilterCompletion)onCompletion;
```

Comment:
In swift it looks like `open func sendAllReports() async throws -> ([any KSCrashReport], Bool)`. Perhaps it would be useful to document the "return value" as well?

Reply:
Don't think it's a right way... Tried playing with the docs but wasn't happy with the results. We can iterate on this later, it's good that API itself looks fine. Maybe we can drop "isSuccess" and assume that "no error == success". This would make API more clean.

Reply:
>  Maybe we can drop "isSuccess" and assume that "no error == success". This would make API more clean.

Yeah, I like this approach

Reply:
Will do separately, it would bring way too many changes to this PR